packit_service/worker/events/event.py

```diff
@@ -110,61 +107,55 @@ def project(self):
-                    repo_name=self.project.repo,
-                    project_url=self.project_url,
-                )
+        return self._db_trigger
 
-            elif self.event_type in {"ReleaseEvent", "CheckRerunReleaseEvent"}:
-                self._db_trigger = ProjectReleaseModel.get_or_create(
-                    tag_name=self.tag_name,
-                    namespace=self.project.namespace,
-                    repo_name=self.project.repo,
-                    project_url=self.project_url,
-                    commit_hash=self.commit_sha,
-                )
-            elif self.event_type in {
-                "IssueCommentEvent",
-                "IssueCommentGitlabEvent",
-            }:
-                self._db_trigger = IssueModel.get_or_create(
-                    issue_id=self.issue_id,
-                    namespace=self.project.namespace,
-                    repo_name=self.project.repo,
-                    project_url=self.project_url,
-                )
-            else:
-                logger.warning(
-                    "We don't know, what to search in the database for this event data."
+    def _get_db_trigger_class(self) -> Optional[_AddDbTrigger]:
+        try:
+            logger.debug(f"Trying to get event type class: {self.event_type}")
+            event_cls = getattr(packit_service.worker.events, self.event_type)
+        except AttributeError:
+            logger.warning(
+                "We don't know, what to search in the database for this event data."
+            )
+            return None
+
+        # first object in list is class itself and last is `object` - let's skip it
+        for superclass in event_cls.mro()[1:-1]:
+            if _AddDbTrigger in superclass.__bases__:
+                logger.debug(
+                    f"AddDbTrigger class found: {superclass.__name__} in {event_cls.mro()}"
                 )
+                return superclass()
 
-        return self._db_trigger
+        logger.debug(f"No AddDbTrigger class found in {event_cls.mro()}")
+        return None
+
+    def _recreate_original_db_trigger(self) -> Optional[AbstractTriggerDbType]:
+        needed_attributes_for_abstract_db_trigger = {
+            "project": self.project,
+            "repo_namespace": self.project.namespace,
+            "repo_name": self.project.repo,
+            "project_url": self.project_url,
+            "pr_id": self.pr_id,
+            "issue_id": self.issue_id,
+            "git_ref": self.git_ref,
+            "tag_name": self.tag_name,
+            "commit_sha": self.commit_sha,
+        }
+        non_none_attributes = {
+            key: value
+            for key, value in needed_attributes_for_abstract_db_trigger.items()
+            if value is not None
+        }
+
+        db_trigger_cls = self._get_db_trigger_class()
+        if db_trigger_cls is None:
+            return None
+
+        db_trigger_cls.__dict__ = non_none_attributes
+        return db_trigger_cls.db_trigger
```

Comment:
The `get_or_create` is a class method so if `_get_db_trigger_class()` returns the base/super class we can call its `get_or_create()` and give it the existing attributes as arguments.

```suggestion
        return db_trigger.get_or_create(**non_none_attributes)
```